server/channels/app/import_functions.go

```diff
@@ -1615,6 +1617,47 @@ func (a *App) importMultiplePostLines(rctx request.CTX, lines []imports.LineImpo
 		if line.Post.IsPinned != nil {
 			post.IsPinned = *line.Post.IsPinned
 		}
+		if line.Post.ThreadFollowers != nil {
+			threadMemberships := []*model.ThreadMembership{}
+			participants := make([]*model.User, len(*line.Post.ThreadFollowers))
+
+			for i, member := range *line.Post.ThreadFollowers {
+				user, ok := users[strings.ToLower(*member.User)]
+				if !ok {
+					// maybe it's a user on target instance but not in the import data.
+					// This is a rare case, but we need or can to handle it.
+					// alternatively, we can continue and discard this follower as maybe they
+					// were deleted.
+					var uErr error
+					user, uErr = a.Srv().Store().User().GetByUsername(*member.User)
+					if uErr != nil {
+						return line.LineNumber, model.NewAppError("importMultiplePostLines", "app.import.get_users_by_username.some_users_not_found.error", nil, "", http.StatusBadRequest).Wrap(uErr)
+					}
+				}
+				membership := &model.ThreadMembership{
+					PostId:    post.Id, // empty if it's a new post, will set later while inserting to the DB.
```

Comment:
Can you clarify in which case the `post.Id` will NOT be empty? It seems to me like it will always be empty until the point you insert it into the DB.

Reply:
As Claudio said, you can check the logic here: https://github.com/mattermost/mattermost/blob/9d816f856da72bf581af51938a5cad88b9a72aee/server/channels/app/import_functions.go#L1585

Reply:
Ah yes ofc :facepalm: